src/Microsoft.Extensions.Logging.Abstractions/LoggerMessage.cs

```diff
@@ -0,0 +1,343 @@
+        /// <typeparam name="T1">The parameter passed to the named format string</typeparam>
+        /// <typeparam name="T2">The parameter passed to the named format string</typeparam>
+        /// <param name="message">A delegate which creates a log message</param>
+        /// <param name="logLevel">The <see cref="LogLevel"/></param>
+        /// <param name="eventId">The event id</param>
+        /// <param name="formatString">The named format string</param>
+        public static void Define<T1, T2>(out Action<ILogger, T1, T2, Exception> message, LogLevel logLevel, int eventId, string formatString)
+        {
+            var formatter = new LogValuesFormatter(formatString);
+
+            message = (logger, arg1, arg2, exception) =>
+            {
+                if (logger.IsEnabled(logLevel))
+                {
+                    logger.Log(logLevel, eventId, new LogValues<T1, T2>(formatter, arg1, arg2), exception, LogValues<T1, T2>.Callback);
+                }
+            };
+        }
+
+        /// <summary>
+        /// Creates a delegate which can be used for logging a message.
+        /// </summary>
+        /// <typeparam name="T1">The parameter passed to the named format string</typeparam>
+        /// <typeparam name="T2">The parameter passed to the named format string</typeparam>
+        /// <typeparam name="T3">The parameter passed to the named format string</typeparam>
+        /// <param name="message">A delegate which creates a log message</param>
+        /// <param name="logLevel">The <see cref="LogLevel"/></param>
+        /// <param name="eventId">The event id</param>
+        /// <param name="formatString">The named format string</param>
+        public static void Define<T1, T2, T3>(out Action<ILogger, T1, T2, T3, Exception> message, LogLevel logLevel, int eventId, string formatString)
+        {
+            var formatter = new LogValuesFormatter(formatString);
+
+            message = (logger, arg1, arg2, arg3, exception) =>
+            {
+                if (logger.IsEnabled(logLevel))
+                {
+                    logger.Log(logLevel, eventId, new LogValues<T1, T2, T3>(formatter, arg1, arg2, arg3), exception, LogValues<T1, T2, T3>.Callback);
+                }
+            };
+        }
+
+        /// <summary>
+        /// Creates a delegate which can be used for logging a message.
+        /// </summary>
+        /// <typeparam name="T1">The parameter passed to the named format string</typeparam>
+        /// <param name="message">A delegate which creates a log message</param>
+        /// <param name="logLevel">The <see cref="LogLevel"/></param>
+        /// <param name="eventId">The event id</param>
+        /// <param name="eventName">The event name</param>
+        /// <param name="formatString">The named format string</param>
+        public static void Define<T1>(out Action<ILogger, T1, Exception> message, LogLevel logLevel, int eventId, string eventName, string formatString)
```

Comment:
I think it's `categoryName` elsewhere: https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging/LoggerFactory.cs#L20